app/code/core/Mage/Sales/Model/Service/Order.php

```diff
@@ -189,7 +189,7 @@ public function prepareShipment($qtys = [])
             } else {
                 if (isset($qtys[$orderItem->getId()])) {
                     $qty = min($qtys[$orderItem->getId()], $orderItem->getQtyToShip());
-                } elseif (!count($qtys)) {
+                } elseif (is_null($qtys) || !count($qtys)) {
```

Comment:
Can be replaced with `empty` which will check both conditions:
```suggestion
                } elseif (empty($qtys)) {
```

```php
$qtys = null;
var_dump(is_null($qtys) || !count($qtys)); // bool(true)
$qtys = [];
var_dump(is_null($qtys) || !count($qtys)); // bool(true)

$qtys = null;
var_dump(empty($qtys)); // bool(true)
$qtys = [];
var_dump(empty($qtys)); // bool(true)
```